build.py

```diff
@@ -59,14 +59,20 @@ def get_os():
     reference = "{0}/{1}".format(name, version)
     upload = "https://api.bintray.com/conan/{0}/public-conan".format(username)
 
+    build = os.getenv("BUILD", None)
+    if build:
+        build_env = os.getenv(build, None)
+        if build_env:
+            for env_var in build_env.split():
+                name, value = env_var.split("=", 1)
+                os.environ[name] = value
+
     builder = ConanMultiPackager(
-        username=username,
-        channel=channel,
-        reference=reference,
-        upload=upload,
-        remotes=upload,  # while redundant, this moves bincrafters remote to position 0
-        upload_only_when_stable=True,
-        stable_branch_pattern="stable/*")
-
-    builder.add_common_builds(shared_option_name=name + ":shared")
+        username=os.getenv("CONAN_USERNAME", username),
+        channel=os.getenv("CONAN_CHANNEL", channel),
+        reference=os.getenv("CONAN_REFERENCE", reference),
+        upload=os.getenv("CONAN_UPLOAD", upload),
+        remotes=os.getenv("CONAN_REMOTES", upload))
+
+    builder.add_common_builds(shared_option_name="*:shared")
```

Comment:
the only problem using this approach is that all packages will be forced to use the same option and will build a specific configuration for such package. As result, when conan install is executed, if some package by default does not use the same option, the package won't be available.

Reply:
Yeah, you are right. But I can't figure out some dynamic way to solve directly in `conanfile.py`. I asked to conan team by an [issue](https://github.com/conan-io/conan/issues/2218)